src/chainstate/burn/db/sortdb.rs

```diff
@@ -1922,13 +2102,35 @@ impl <'a> SortitionHandleTx <'a> {
 
         // if this commit has a sortition, record its burn block hash and stacks block hash
         if snapshot.sortition {
-            keys.push(db_keys::last_sortition());
+            keys.push(db_keys::last_sortition().to_string());
             values.push(snapshot.burn_header_hash.to_hex());
 
             keys.push(db_keys::stacks_block_present(&snapshot.winning_stacks_block_hash));
             values.push(snapshot.sortition_id.to_hex());
         }
 
+        // if this is the start of a reward cycle, store the new PoX keys
+        if !snapshot.is_initial() {
+            if let Some(reward_info) = next_pox_info {
+                let mut pox_id = self.get_pox_id()?;
+                if reward_info.anchor_block_known {
+                    pox_id.extend_with_present_block();
+                } else {
+                    pox_id.extend_with_not_present_block();
+                }
+                if let Some(ref anchor_block) = reward_info.anchor_block {
```

Comment:
Why have a separate `anchor_block_known` field if we appear to be able to deduce whether or not we know of the anchor block using whether or not `reward_info.anchor_block` is `Some` or `None`?

Reply:
If an anchor block isn't selected (e.g., there's two forks each with 50% of the confirmations during the prepare phase), then you'd have `anchor_block = None`, but `anchor_block_known = true`

Reply:
But, the anchor block would not be known, because there is no anchor block in that scenario?

Reply:
Yes, but the resulting reward cycle is the only possible valid reward cycle, which is the important distinction when it comes to processing the sortitions. In the PoX fork handling, a "non-present anchor block" indicates that a subsequently arriving Stacks block could cause this PoX fork to become invalid. If no anchor is selected, that's not the case.

Reply:
Ah.  So, is the following interpretation correct?

* If `anchor_block_known == false`, then there is no anchor block in this reward cycle, _and there can never be one_.
* If `anchor_block_known == true` and `reward_info.anchor_block` is `None`, then a Stacks block hash got enough confirmations when looking at the burnchain, but we do not yet have the associated anchor block (so we don't know if the hash corresponds to a valid Stacks block yet).
* If `anchor_block_known == true` and `reward_info.anchor_block` is `Some(..)`, then we know for a fact that valid Stacks block has been chosen as the anchor block.  The canonical PoX fork will be updated.

Reply:
Thanks!